### PR TITLE
fix: Use JSX.ImplicitElements to derive valid property names

### PR DESCRIPTION
**Pull request by ahobson**
# Summary

Use the proper react types to indicate which props are allowed on components.

## Related Issues or PRs
This was found as part of exploration for #161

## How To Test
This should not change functionality other than allowing the proper attributes

### CODE REVIEW

File: src/components/header/MegaMenu/MegaMenu.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const MegaMenu = (
       <div className="grid-row grid-gap-4">
         {items.map((listItems, i) => (
           <div className="usa-col" key={`subnav_col_${i}`}>
-            <NavList items={listItems} type="megamenu" {...ulProps} />
+            <NavList items={listItems} megamenu={true} {...navListProps} />
```
Comment (haworku) [inline src/components/header/MegaMenu/MegaMenu.tsx:22]: suggestion:  for `megamenu={true}` and I think I saw this one other place.  Leave off the attribute assignment when its `true.` Just by including the name of the attribute on the element it is by default true.

File: src/components/Tag/Tag.tsx [hunk 2]
```diff
@@ -19,7 +19,11 @@ export const Tag = (
   const tagClasses = classnames('usa-tag', className)
 
   return (
-    <span data-testid="tag" className={tagClasses} style={{ ...style }}>
+    <span
+      data-testid="tag"
+      className={tagClasses}
+      style={{ ...style }}
+      {...spanProps}>
```
Comment (haworku) [inline src/components/Tag/Tag.tsx:26]: Thanks for catching this, I think I saw a couple other places where this rewrite was kind of a cleanup for us.

**Comment by ahobson** [comment]
> Looks good to me! Just need to take [WIP] out of the PR title and make it match the semantic changelog format.

Ok!  Should I create a corresponding issue about it?  Is this a bugfix or a feature?

**Comment by haworku** [comment]
@ahobson I think `fix` is fine with the same PR title you have.  Also don't feel a need to make an issue since this came up in #161 somewhat organically.

**Pull request merged**